tracer/src/Datadog.Trace/Iast/EvidenceConverter.cs

```diff
@@ -59,29 +59,34 @@ private static string Substring(string value, Range range)
         return value.Substring(range.Start, range.Length);
     }
 
-    public override Evidence ReadJson(JsonReader reader, Type objectType, Evidence existingValue, bool hasExistingValue, JsonSerializer serializer)
+    public override Evidence? ReadJson(JsonReader reader, Type objectType, Evidence? existingValue, bool hasExistingValue, JsonSerializer serializer)
     {
         throw new NotImplementedException();
     }
 
-    public override void WriteJson(JsonWriter writer, Evidence evidence, JsonSerializer serializer)
+    public override void WriteJson(JsonWriter writer, Evidence? evidence, JsonSerializer serializer)
     {
+        if (evidence is null)
+        {
+            return;
+        }
```

Comment:
Is this correct, you don't want to write `null` into the document? It seems like this could potentially create invalid json?

Reply:
This case should never happen actually. I added that code to make sure that we don't run into a NullReferenceException. The evidence converter should only be called when there is actually an evidence.

Reply:
That makes sense, but _given_ this will never be called, I still kind of think you should do this to be completely safe 😄 

```suggestion
        if (evidence is null)
        {
            writer.WriteNull();
            return;
        }
```